File: codegen/codegen-test/src/test/resources/golden/elixir/restjson1/server/weather_server_impl.ex
defmodule Weather.Server.Impl do
  @moduledoc false
  @behaviour Weather.Server.Handler

  # This file will NOT be overwritten. Add your business logic here.

  alias Weather.Server.Types.GetWeatherInput
  alias Weather.Server.Types.GetWeatherOutput
  alias Weather.Server.Types.ListCitiesInput
  alias Weather.Server.Types.ListCitiesOutput

  @impl true
  def get_weather(%GetWeatherInput{} = _input, _ctx), do: {:error, :not_implemented}

  @impl true
  def list_cities(%ListCitiesInput{} = _input, _ctx), do: {:error, :not_implemented}
end

File: codegen/codegen-test/src/test/resources/golden/erlang/restjson1/server/weather_server_impl.erl
-module(weather_server_impl).

%% This file will NOT be overwritten. Add your business logic here.
-behaviour(weather_server).

-export([
    get_weather/2,
    list_cities/2
]).

-spec get_weather(weather_server:get_weather_input(), map()) ->
    {ok, weather_server:get_weather_output()} | {error, term()}.
get_weather(_Input, _Context) ->
    {error, not_implemented}.

-spec list_cities(weather_server:list_cities_input(), map()) ->
    {ok, weather_server:list_cities_output()} | {error, term()}.
list_cities(_Input, _Context) ->
    {error, not_implemented}.
